vms/job/tests/test_model.py

```diff
@@ -0,0 +1,117 @@
+# third party
+
+# Django
+from django.db.models import Q
+from django.test.testcases import TestCase
+
+# local Django
+from job.models import Job
+from event.models import Event
+from shift.utils import create_job_with_details, create_event_with_details
+
+
+class JobModelTests(TestCase):
+
+    def setUp(self):
+        self.event = self.create_event()
+
+    def tearDown(self):
+        pass
+
+    @staticmethod
+    def create_event():
+        event = ['event-name', '2050-05-24', '2050-05-28']
+        created_event = create_event_with_details(event)
+        return created_event
+
+    def create_job(self):
+        job = ['job-name', '2050-05-25', '2050-05-26', 'job-description', self.event]
+        created_job = create_job_with_details(job)
+        return created_job
+
+    def test_valid_model_create(self):
+        job = ['job-name', '2050-05-25', '2050-05-26', 'job-description', self.event]
+        create_job_with_details(job)
+
+        # Check database for instance creation
+        self.assertNotEqual(len(Job.objects.all()), 0)
+
+        event_in_db = Event.objects.get(Q(name='event-name'))
+        # Check if event name is correct
+        self.assertEqual(event_in_db.name, self.event.name)
+
+        job_in_db = Job.objects.get(Q(event=event_in_db))
+        # Check if job details are correct
+        self.assertEqual(job_in_db.name, job[0])
+        self.assertEqual(str(job_in_db.start_date), job[1])
+        self.assertEqual(str(job_in_db.end_date), job[2])
+        self.assertEqual(job_in_db.description, job[3])
+
+    # Checks are missing from model class
```

Comment:
Raise issue after confirmation in tomorrow's meeting.

Reply:
Workaround found:
model validation checks aren't performed until a `full_clean` is called explicitly from code/command-line, Forms perform `full_clean` on submit thats why the error messages are shown. Thus `model.save()` will save the instance unless there not-null constraint is broken.

Reply:
Its resolved, by mistake I wrote workaround found instead should be Solution found